signature/src/image/mod.rs

```diff
@@ -75,53 +77,25 @@ impl Image {
         }
     }
 
-    pub fn signatures(&mut self) -> Result<Vec<Vec<u8>>> {
+    pub fn signatures(&mut self, scheme: &str) -> Result<Vec<Vec<u8>>> {
```

Comment:
This is a rather odd method as `scheme` is ignored if it's already been called. It might be clearer to have a "setter method" that accepts the `scheme` and then this can remain parameter free (and error if the setter hasn't been called?

Reply:
Here, we probably shouldn't store `signatures` and `sigstore_base_url` as the member fields of image, because an image may be required to verify the signatures under different schemes. I updated the code accordingly.

Reply:
Here, we probably shouldn't store `signature` and `sigstore_base_url` as the member fields of `image`, because an image may be required to verify the signatures under different schemes. I updated the code accordingly.